src/test/java/eu/markov/jenkins/plugin/mvnmeta/MavenMetadataParameterDefinitionTest.java

```diff
@@ -137,11 +139,46 @@ public void testListVersionsSingleSnapshot() {
 
     @Test
     public void testListVersionsTimestampedSnapshot() {
-        MavenMetadataParameterDefinition definition = new MavenMetadataParameterDefinition("variable", "void", getLocalWebServerUrl(), "com.acme", "timestamped", "jar", "", "DESC", "null", "2", "", "");
+        MavenMetadataParameterDefinition definition = new MavenMetadataParameterDefinition("variable", "void", getLocalWebServerUrl(), "com.acme", "timestamped", "jar", "", "DESC", "null", "2", "", "", "", "", "");
         List<String> versions = definition.getVersions();
         Assert.notEmpty(versions);
         assertEquals(2, versions.size());
         assertEquals("3.8-SNAPSHOT", versions.get(0));
     }
 
-}
\ No newline at end of file
+    @Test
+    public void testGetCurrentArtifactInfoPatternWithCapturingGroup() {
+        this.testGetCurrentArtifactInfoPattern(CURRENT_ARTIFACT_INFO_URL, "My Label", "Plugin Version: ([\\S]+)", "My Label: 3.14159");
+    }
+
+    @Test
+    public void testGetCurrentArtifactInfoPatternWithoutCapturingGroup() {
+        this.testGetCurrentArtifactInfoPattern(CURRENT_ARTIFACT_INFO_URL, "My Label", "Plugin Version: [\\S]+", "My Label: Plugin Version: 3.14159");
+    }
+
+    @Test
+    public void testGetCurrentArtifactInfoWithNonmatchingPattern() {
+        this.testGetCurrentArtifactInfoPattern(CURRENT_ARTIFACT_INFO_URL, "My Label", "Not Matching: [\\S]+", "My Label: Artifact ID: my-artifact\nPlugin Version: 3.14159\nJenkins Version:  42\n");
+    }
+
+    @Test
+    public void testGetCurrentArtifactInfoWithoutPattern() {
+        this.testGetCurrentArtifactInfoPattern(CURRENT_ARTIFACT_INFO_URL, "My Label", null, "My Label: Artifact ID: my-artifact\nPlugin Version: 3.14159\nJenkins Version:  42\n");
+    }
+
+    @Test
+    public void testGetCurrentArtifactInfoPatternWithDefaultLabel() {
+        this.testGetCurrentArtifactInfoPattern(CURRENT_ARTIFACT_INFO_URL, null, "Plugin Version: ([\\S]+)", "Currently used artifact: 3.14159");
+    }
+
+    @Test
+    public void testGetCurrentArtifactInfoPatternWithoutUrl() {
+        this.testGetCurrentArtifactInfoPattern(null, "My Label", "Plugin Version: ([\\S]+)", "");
+    }
+
+    private void testGetCurrentArtifactInfoPattern(String url, String label, String pattern, String expectedResult) {
```

Comment:
usually only tests starting with test prefix for method, if it just a helper function, then better make it static and remove test prefix.